### PR TITLE
Allow using collection properties to configure crossdc updatehandler.

### PR DESCRIPTION
**Pull request by markrmiller**

### CODE REVIEW

File: crossdc-consumer/build.gradle [hunk 1]
```diff
@@ -60,8 +60,8 @@ dependencies {
         exclude group: "org.apache.logging.log4j", module: "*"
         exclude group: "org.slf4j", module: "*"
     }
-    testImplementation 'org.apache.kafka:kafka_2.13:2.8.1'
```
Comment (anshumg) [inline crossdc-consumer/build.gradle:63]: this change seems unrelated

File: crossdc-producer/src/main/java/org/apache/solr/update/processor/MirroringUpdateRequestProcessorFactory.java [hunk 2]
```diff
@@ -152,6 +149,22 @@ public void inform(SolrCore core) {
 
                 KafkaCrossDcConf.readZkProps(properties, zkProps);
             }
+            CollectionProperties cp = new CollectionProperties(core.getCoreContainer().getZkController().getZkClient());
```
Comment (anshumg) [inline crossdc-producer/src/main/java/org/apache/solr/update/processor/MirroringUpdateRequestProcessorFactory.java:152]: Let's extract this out into a method? `isCrossDcEnabled()` or something?

File: crossdc-producer/src/test/java/org/apache/solr/crossdc/SolrAndKafkaMultiCollectionIntegrationTest.java [hunk 3]
```diff
@@ -169,6 +181,14 @@ public void testFullCloudToCloudMultiCollection() throws Exception {
       solrCluster2.getSolrClient().request(create);
       solrCluster2.waitForActiveCollection(ALT_COLLECTION, 1, 1);
 
+      // Update the collection property "enabled" to true
+      CollectionProperties cp = new CollectionProperties(solrCluster1.getZkClient());
+      cp.setCollectionProperty(ALT_COLLECTION, "crossdc.enabled", "true");
+      cp.setCollectionProperty(ALT_COLLECTION, "crossdc.topicName", TOPIC);
```
Comment (anshumg) [inline crossdc-producer/src/test/java/org/apache/solr/crossdc/SolrAndKafkaMultiCollectionIntegrationTest.java:187]: It seems like we're testing the fetching of the correct topic name, something that we can change at run time but I'm viewing this on the browser. Can you confirm?

**Review by anshumg — APPROVED** [review]
Left a few suggestions but looks good overall. The one thing that we must make sure is that the topic name is fetched on collection reload i.e. as part of the URP Factory.

File: crossdc-consumer/build.gradle (hunk 1 above)
Comment (markrmiller) [reply crossdc-consumer/build.gradle:63]: It's to fix being able run properly in the ide, which is related to debugging.

File: crossdc-producer/src/test/java/org/apache/solr/crossdc/SolrAndKafkaMultiCollectionIntegrationTest.java (hunk 3 above)
Comment (markrmiller) [reply crossdc-producer/src/test/java/org/apache/solr/crossdc/SolrAndKafkaMultiCollectionIntegrationTest.java:187]: We just checking that properies besides enable (which is handled specially) also work by moving this from a system property to a collection property.

File: crossdc-producer/src/main/java/org/apache/solr/update/processor/MirroringUpdateRequestProcessorFactory.java [hunk 4]
```diff
@@ -152,6 +149,22 @@ public void inform(SolrCore core) {
...
+             Map<String,String> collectionProperties = cp.getCollectionProperties(core.getCoreDescriptor().getCollectionName());
+            for (ConfigProperty configKey : KafkaCrossDcConf.CONFIG_PROPERTIES) {
+                String val = collectionProperties.get("crossdc." + configKey.getKey());
+                if (val != null && !val.isBlank()) {
+                    properties.put(configKey.getKey(), val);
+                }
+            }
+            String enabledVal = collectionProperties.get("crossdc.enabled");
```
Comment (gerlowskija) [inline crossdc-producer/src/main/java/org/apache/solr/update/processor/MirroringUpdateRequestProcessorFactory.java:160]: [0] We should probably document these special collection properties in Javadocs at the least, right?

**Comment by gerlowskija** [comment]
LGTM on the whole.
